neuralop/data/transforms/rescale.py

```diff
@@ -0,0 +1,372 @@
+    """
+    unsqueeze_x = x.ndim == 3  # check whether batch dimension is missing
+    unsqueeze_y = y.ndim == 3
+
+    # add batch dimension
+    if unsqueeze_x:
+        x = x.unsqueeze(0)
+    if unsqueeze_y:
+        y = y.unsqueeze(0)
+
+    # extract BC
+    Nx, Ny = y.shape[-2], y.shape[-1]
+    x[:, 1:3, :, :] = y[:, :, 0, :].unsqueeze(-2).repeat(1, 1, Nx, 1)  # left boundary
+    x[:, 3:5, :, :] = y[:, :, :, 0].unsqueeze(-1).repeat(1, 1, 1, Ny)  # lower boundary
+    x[:, 5:7, :, :] = y[:, :, -1, :].unsqueeze(-2).repeat(1, 1, Nx, 1)  # right boundary
+    x[:, 7:9, :, :] = y[:, :, :, -1].unsqueeze(-1).repeat(1, 1, 1, Ny)  # upper boundary
+
+    # undo adding of batch dimension
+    if unsqueeze_x:
+        x = x[0]
+    if unsqueeze_y:
+        y = y[0]
+
+    return x
+
+
+def GridResize(x, grid_size, mode="bilinear"):
+    """
+    Resize the grid values by interpolation in the last two components.
+    Expected input is either of size
+        batch x channel x original_size x original_size
+    or
+        channel x original_size x original_size
+    """
+    if x.ndim == 4:
+        return torch.nn.functional.interpolate(
+            x, size=(grid_size, grid_size), mode="bilinear", align_corners=True
+        )
+    elif x.ndim == 3:
+        return torch.nn.functional.interpolate(
+            x.unsqueeze(0),
+            size=(grid_size, grid_size),
+            mode="bilinear",
+            align_corners=True,
+        ).squeeze(0)
+    else:
+        ValueError(
+            f"Input x to GridResize must be a tensor with either 3 or 4 dimensions! {x.ndims=}, {x.shape=}"
+        )
+
+
+# data augmentation routines
+class GridResizing:
+    def __init__(self, grid_size, ExtractBC=DarcyExtractBC):
+        self.grid_size = grid_size
+
+    def __call__(self, x, y):
+        return self.forward(x, y)
+
+    def forward(self, x, y):
+        x = GridResize(x, self.grid_size)
+        y = GridResize(y, self.grid_size)
+        x = ExtractBC(x, y)
+        return x, y
+
+
+class RandomFlip:
+    def __init__(self, p=0.5, ExtractBC=DarcyExtractBC):
+        self.p = p
+        self.ExtractBC = ExtractBC
+
+    def __call__(self, x, y):
+        return self.forward(x, y)
+
+    def forward(self, x, y):
+        if torch.rand(1) < self.p:
+            x, y = F.hflip(x), F.hflip(y)
+        if torch.rand(1) < self.p:
+            x, y = F.vflip(x), F.vflip(y)
+        if torch.rand(1) < self.p:
+            x, y = x.transpose(-1, -2), y.transpose(-1, -2)
+        x = self.ExtractBC(x, y)
+        return x, y
+
+
+class RandomRotation:
+    def __init__(self, p=0.5, ExtractBC=DarcyExtractBC):
+        raise NotImplementedError
+
+    def __call__(self, x, y):
+        return self.forward(x, y)
+
+    def forward(self, x, y):
+        if torch.rand(1) < self.p:
+            x, y = x.transpose(-1, -2), y.transpose(-1, -2)
+        x = ExtractBC(x, y)
+        return x, y
+
+
+class RandomCropResize:
```

Comment:
Is there a specific reason we can't reshape and use `torchvision.transforms` directly?